### PR TITLE
randomx: 1.1.9 -> 1.1.10

### PR DESCRIPTION
**Pull request by viraptor**
###### Description of changes

New sdk is needed for `pthread_jit_write_protect_np`. Replaces https://github.com/NixOS/nixpkgs/pull/186502

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20994,7 +20994,7 @@ with pkgs;
 
   qgnomeplatform = libsForQt5.callPackage ../development/libraries/qgnomeplatform { };
 
-  randomx = callPackage ../development/libraries/randomx { };
+  randomx = darwin.apple_sdk_11_0.callPackage ../development/libraries/randomx { };
```
Comment (viraptor) [inline pkgs/top-level/all-packages.nix:20997]: Required to access `pthread_jit_write_protect_np`. See https://github.com/NixOS/nixpkgs/pull/186502#issuecomment-1214310441

Comment (rnhmjoj) [reply pkgs/top-level/all-packages.nix:20997]: Not that it's your fault, but I don't really like this design: a particular platform shouldn't "take over" the package definition, also it seems it doesn't compose.

Comment (rnhmjoj) [reply pkgs/top-level/all-packages.nix:20997]: By the way, your comment should be a comment in the source code.

Comment (viraptor) [reply pkgs/top-level/all-packages.nix:20997]: It's a hopefully temporary fix while the sdk issues are being resolved https://github.com/NixOS/nixpkgs/pull/176661 but yeah, for now we need to put up with some imprefect solutions.

**Review by rnhmjoj — APPROVED** [review]
I built and tested monero and monero-gui: everything looks ok.

**Pull request merged**